lib/uxtx.js

```diff
@@ -0,0 +1,625 @@
+const { Amount, TX } = require('bcoin');
+const assert = require('bsert');
+const moment = require('moment');
+
+/**
+ * User Experience Transaction
+ * NOTE: this class is not meant to be mutable
+ *
+ * @alias module:bpanel-utils.UXTX
+ * @extends TX
+ */
+class UXTX extends TX {
+  /**
+   * Create a UXTX
+   * User Experience Transaction
+   * @constructor
+   * @param options
+   *
+   * TODO: handle these cases:
+   * wallet/account -> same wallet/account
+   * wallet/account -> same wallet/different account
+   * coinjoin transactions
+   *
+   * TODO: overwrite TX.inspect
+   */
+
+  constructor(options) {
+    super();
+
+    // these are used as keys to look up
+    // in options.labels for toJSON labels
+    this.TYPES = {
+      COINBASE: 'COINBASE',
+      WITHDRAW: 'WITHDRAW',
+      DEPOSIT: 'DEPOSIT',
+      UNKNOWN: 'UNKNOWN',
+    }
+
+    this.DATE_FORMAT = null;
+    this._UXType = null;
+    this._labels = null;
+    this._json = null;
+    this._wallet = null;
+    this._account = null;
+    this._accounts = null;
+    this._counterparty = null;
+    this._recipients = null;
+
+    if (options)
+      this.fromOptions(options);
+  }
+
+  /**
+   * Inject properties from options object.
+   * @private
+   * @param {Object} options
+   */
+  fromOptions(options) {
```

Comment:
Would it make sense to maintain the parent's instantiation from options? That way you could pass normal tx options in an object that also includes these UX specific ones? I'm not sure, and am indifferent I think, just wondering if that adds more utility.

Reply:
That could be useful, I originally wanted `fromJSON` to be the best way to instantiate this class, but it turns out that the `bcoin` wallet api doesn't return enough information to instantiate `TX` class from the json it returns, but it does return the `tx` hex, which allows you to instantiate with `.fromRaw`. I created `fromOptions` to follow the design patterns common in `bcoin`. Maybe we can do something like:

```bash
super.fromOptions(options);
```

if it doesn't break anything? Then we will be extending it and still be able to set custom options and accept any other options we decide are useful

Reply:
Yeah, that sounds good to me.